### ISSUE TITLE
Here's an unexpected error when I use the ```use``` keyword

### ISSUE THREAD

**Issue by 3442853561** [rust-lang/rust#49011]
Here's an unexpected error when I use the ```use``` keyword. This program can run

```Rust
mod foo {
    pub fn Box<T>(t: T) -> Box<T> {
        Box::new(t)
    }
}

fn main() {
    let Box = foo::Box(5);
    let Box = foo::Box(Box);
    println!("{}", Box);
}
```
However, when I use the ```use``` keyword

```Rust
mod foo {
    pub fn Box<T>(t: T) -> Box<T> {
        Box::new(t)
    }
}

use foo::Box;

fn main() {
    let Box = Box(5);
    let Box = Box(Box);
    println!("{}", Box);
}
```

It can not run.

And this can run:

```Rust
mod foo {
    pub fn Box<T>(t: T) -> Box<T> {
        Box::new(t)
    }
}

use foo::Box;

fn main() {
    let Box = foo::Box(5);
    let Box = foo::Box(Box);
    println!("{}", Box);
}
```

and this also can run

```Rust
mod foo {
    pub fn Box<T>(t: T) -> Box<T> {
        Box::new(t)
    }
}

use foo::Box;

fn main() {
    let Box = Box(5);
    let Box = foo::Box(Box);
    println!("{}", Box);
}
```

https://github.com/rust-lang/rust/issues/49009
